src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -66,7 +75,30 @@ const CriteriaView = ({ criteria, deleteCriteria }: CriteriaViewProps) => {
               case 'domain':
                 return h(Fragment, [strong([`${criteria.domainOption.category}:`]), ` ${criteria.name}`]);
               case 'list':
-                return h(Fragment, [strong([`${criteria.name}:`]), ` ${criteria.value.name}`]);
+                return h(Fragment, [
+                  strong([`${criteria.name}: ${_.flow(_.map('name'), _.join(','))(criteria.valuesSelected)}`]),
+                  h(Select, {
+                    'aria-label': `Select one or more ${criteria.name}`,
+                    isClearable: true,
+                    isMulti: true,
+                    options: _.map((value) => ({ label: value.name, value: value.id }), criteria.listOption.values),
+                    value: _.map((value) => ({ label: value.name, value: value.id }), criteria.valuesSelected),
+                    onChange: (values) =>
+                      updateCriteria(
+                        _.set(
+                          'valuesSelected',
+                          _.flow(
+                            _.map((value: Option) => ({
+                              name: value.label,
+                              id: value.value,
+                            })),
+                            _.xorWith(_.isEqual, criteria.valuesSelected)
+                          )((values as Option[]) || null),
+                          criteria
+                        )
+                      ),
```

Comment:
If you wanted, I think you can get rid of the nested function call here by using `flow`
```suggestion
                      _.flow(
                        _.set(
                          'valuesSelected',
                          _.flow(
                            _.map((value: Option) => ({
                              name: value.label,
                              id: value.value,
                            })),
                            _.xorWith(_.isEqual, criteria.valuesSelected)
                          )((values as Option[]) || null),
                        ),
                         updateCriteria
                      )(criteria),
```